### ISSUE TITLE
Use rebuild framework to unwrap tabling

### ISSUE THREAD

**Issue by jburse** [jburse/jekejeke-devel#188]
We already managed to hide the wrapper, the cache
and the lock from tabling. This is useful when large caches
are created, like calling fib(1000,X). But then when we list

there is still the auxiliary name of the predicate:
```
:- sys_notrace fib_a/2.
fib_a(0, X) :-
   !,
   X = 1.
fib_a(1, X) :-
   !,
   X = 1.
fib_a(N, X) :-
   H is N-1,
   fib(H, Y),
   J is H-1,
   fib(J, Z),
   X is Y+Z.
```

**Comment by jburse** [comment]
We keep this ticket here open, whereas we closed the
other ticket here. We closed the other ticket here, since
the other ticket insuinates a more broader application

of unexpansion. So this ticket put on hold via closed:

**Harden wrapper-less un-expansion #200**
[https://\github.com/jburse/jekejeke-devel/issues/200](https://github.com/jburse/jekejeke-devel/issues/200)

Concerning the tabling problem we are rather faced with
a solution to un-expand predicate declarations. We might
consider during un-expand to have [] and '.' as well.

The [] could be used to suppress an un-expansion result.

**Comment by jburse** [comment]
We used our DatalogF3 prototype to explore rebuild framework.
We already introduced a new flag sys_untabled. Without rebuild
and the flag sys_untabled the following is seen:
```
p(A) :-
   p_a(A).

p_a(just(a)).
p_a(nothing).
p_a(just(b)).

q(A) :-
   q_a(A).

q_a(X) :-
   p(X),
   p(Y),
   sql_eq(X, Y).
```
The stubs can be hidden by the flag automatic which we introduced
for the Java class auto loader. The rebuild and the flag sys_untabled
the following is seen:
```
p(just(just(a))).
p(just(nothing)).
p(just(just(b))).

q(X) :-
   p(X),
   p(Y),
   sql_eq(X, Y).
```
We could even untranslate DatalogF3. But as a first proof of concept
for rebuild framework its already ok.

**Comment by jburse** [comment]
A synergistic name finding for new more generic predicate properties
that could be also used by abolish/1 and listing/1 would be to adopt
the following corresponding operator properties:

- **sys_alias:**
  From operator sys_alias/1, just an atom. For predicates could
  assume same arity as well. But this is not clear. To help
  expansion, probably need sys_alias/2 with further info.

- **sys_portray:**
   From operator sys_portray/1, just an atom. For predicates could
   assume same arity as well. But this is not clear. To help
   unexpansion, probably need sys_portray/2 with further info.

- **directive:**
  Instead of supressing the stub via atomatic, try regenerating
  the directive. Prototype the behaviour in DatalogF3.

If we dont help expansion, only the auxiliary name can guide
us, but this name is not a reliable source to guide expansion.
We could use a random atom.

While working for the above new requirements, we could as
well in parallel already replicate the new sys_untabled/0 from
DatalogF3 prototype to unrelated library(tabling).
